### PR TITLE
Kubernetes 1.5.2 rebase

### PR DESCRIPTION
**Pull request by soltysh**
@ncdc @deads2k @liggitt @smarterclayton ptal
@derekwaynecarr ptal the reverted UPSTREAM prs you asked for

The k8s branch is here: https://github.com/soltysh/kubernetes/tree/openshift-1.5

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

**Comment by deads2k** [comment]
wow, that went amazingly well.

**Comment by soltysh** [comment]
> wow, that went amazingly well.

aside from the fact that I can't `go get go4.org/errorutil` today, but it worked yesterday somehow - yes 😃

**Comment by ncdc** [comment]
This isn't the first time go4.org has had issues

**Comment by soltysh** [comment]
Updated commit messages for the first revert which failed commitchecker.

**Comment by liggitt** [comment]
Want to pick https://github.com/kubernetes/kubernetes/issues/39848 or shall I after this merges?

**Comment by soltysh** [comment]
> Want to pick kubernetes/kubernetes#39848 or shall I after this merges?

I can add it here.

**Comment by ncdc** [comment]
I thought about it more, and I think we do need an openshift-1.5.2 branch in our kube fork, so we don't end up force pushing to the openshift-1.5 branch. Or we need to push a tag for the current HEAD of openshift-1.5, so it's still a valid sha.

**Comment by liggitt** [comment]
we've typically tagged and force-pushed the release branch in the kube fork

**Comment by ncdc** [comment]
That's fine, as long as there is a tag.

**Comment by soltysh** [comment]
This push includes @liggitt request upstream PR and fixes the generated changes. Hopefully this should be good.

**Comment by ncdc** [comment]
```
In suite "github.com/openshift/origin/test/cmd/cmd-admin-namespace-setup", test case "hack/test-cmd.sh:257: executing 'oc new-project 'cmd-admin'' expecting success" failed:
=== BEGIN TEST CASE ===
hack/test-cmd.sh:257: executing 'oc new-project 'cmd-admin'' expecting success
FAILURE after 0.471s: hack/test-cmd.sh:257: executing 'oc new-project 'cmd-admin'' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
Error from server: no Items field in &unversioned.Status{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ListMeta:unversioned.ListMeta{SelfLink:"", ResourceVersion:""}, Status:"Success", Message:"", Reason:"", Details:(*unversioned.StatusDetails)(nil), Code:0}
```

**Comment by ncdc** [comment]
Need to pick https://github.com/kubernetes/kubernetes/pull/39886

**Comment by soltysh** [comment]
Rebased, with 39886 picked and updated generated changes.

**Comment by soltysh** [comment]
[testextended][extended:all]

**Comment by soltysh** [comment]
[testextended][extended:all]

**Comment by soltysh** [comment]
Flake #11016 on StatefulSets, need to double check the rest.

[testextended][extended:all]

**Comment by soltysh** [comment]
I've rebased on top of master and added the test for `ExperimentalKernelMemcgNotification` being false, as Clayton asked for it. @smarterclayton are you ok with merging this on green tests? I'd really like to not wait too much longer to deliver this 😄

**Comment by ncdc** [comment]
+1

**Comment by soltysh** [comment]
[(8) Rebase (kube 1.5)](https://trello.com/c/SWNsQvSx/766-8-rebase-kube-1-5)

**Comment by soltysh** [comment]
The tests are green, the only failures in extended tests I'm seeing here are also happening in master. I've opened #12499, #12500 and #12501 covering those. Are there any objections in merging this in @smarterclayton @liggitt @deads2k @ncdc ?

**Comment by ncdc** [comment]
Merge gets my vote.

File: pkg/cmd/server/kubernetes/master.go [hunk 1]
```diff
@@ -118,6 +118,8 @@ func (c *MasterConfig) RunPersistentVolumeAttachDetachController(client *kclient
 			c.CloudProvider,
 			kctrlmgr.ProbeAttachableVolumePlugins(s.VolumeConfiguration),
 			nil,
+			s.DisableAttachDetachReconcilerSync,
```
Comment (deads2k) [inline pkg/cmd/server/kubernetes/master.go]: What are the defaults for these?

**Comment by deads2k** [comment]
[merge]

File: pkg/cmd/server/kubernetes/master.go (hunk 1 above)
Comment (soltysh) [inline pkg/cmd/server/kubernetes/master.go]: `ReconcilerSyncLoopPeriod` is 5 sec, `DisableAttachDetachReconcilerSync` is false. The former is tested in pkg/cmd/server/kubernetes/master_config_test.go, I'll add one another for the latter in that case as a followup.

**Pull request merged**